### PR TITLE
Add Hello World example

### PR DESCRIPTION
**Pull request by chalin**
@lowasser could you review this PR? /cc @bshaffer @thisisnotapril

### CODE REVIEW

File: example/src/main/proto/helloworld.proto [hunk 1]
```diff
@@ -0,0 +1,38 @@
+// Copyright 2015 gRPC authors.
```
Comment (bshaffer) [inline example/src/main/proto/helloworld.proto]: 2015 called, it wants its copyright header back :P

File: example/src/main/kotlin/io/grpc/examples/helloworld/HelloWorldClient.kt [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+import java.util.concurrent.TimeUnit
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.asCoroutineDispatcher
+import kotlinx.coroutines.asExecutor
+import kotlinx.coroutines.runBlocking
+
+class HelloWorldClient constructor(
+    private val channel: ManagedChannel
+) : Closeable {
+  private val stub: GreeterCoroutineStub = GreeterCoroutineStub(channel);
```
Comment (lowasser) [inline example/src/main/kotlin/io/grpc/examples/helloworld/HelloWorldClient.kt]: Skip the semicolon.

File: example/src/main/kotlin/io/grpc/examples/helloworld/HelloWorldClient.kt [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+fun main(args: Array<String>) {
+  val port = 50051
+
+  Executors.newFixedThreadPool(10).asCoroutineDispatcher().use { dispatcher ->
+    HelloWorldClient(
+        ManagedChannelBuilder.forAddress("localhost", port)
+            .usePlaintext()
+            .executor(dispatcher.asExecutor()).build()
+    ).use {
+      val user = if (args.size > 0) args[0] else "world"
```
Comment (lowasser) [inline example/src/main/kotlin/io/grpc/examples/helloworld/HelloWorldClient.kt]: I'd tend to write `val user = args.singleOrNull() ?: "world"`, which gives an error if there is more than one argument.

File: example/src/main/kotlin/io/grpc/examples/helloworld/HelloWorldClient.kt (hunk 2 above)
Comment (chalin) [reply example/src/main/kotlin/io/grpc/examples/helloworld/HelloWorldClient.kt]: Oops, done!

File: example/src/main/kotlin/io/grpc/examples/helloworld/HelloWorldClient.kt (hunk 3 above)
Comment (chalin) [reply example/src/main/kotlin/io/grpc/examples/helloworld/HelloWorldClient.kt]: Yes! Done.

**Pull request merged**